frontend/src/main/java/com/google/tripmeout/frontend/servlet/ServletsModule.java

```diff
@@ -9,9 +9,9 @@
 public class ServletsModule extends ServletModule {
   @Override
   protected void configureServlets() {
-    serve("/api/trips/*/placeVisits/*").with(PlaceVisitIndividualServlet.class);
-    serve("/api/trips/*/placeVisits").with(PlaceVisitParentServlet.class);
-    serve("/api/trips/*").with(TripServlet.class);
+    serveRegex("/api/trips/[^/]+/placeVisits/[^/]+").with(PlaceVisitIndividualServlet.class);
+    serveRegex("/api/trips/[^/]+/placeVisits").with(PlaceVisitParentServlet.class);
+    serveRegex("/api/trips/[^/]+").with(TripServlet.class);
     serve("/api/trips").with(TripParentServlet.class);
```

Comment:
We might want to consider adding an explicit anchor to the end of this. Otherwise, I'm guessing that subpaths that might otherwise match the above servlets (e.g., `/api/trips/mytrip/placeVisits/myvisit/bad/subpath`) will get routed to `TripParentServlet`. In that case, it seems appropriate to return a bad-request response code rather than silently use the wrong servlet.

Reply:
Just tested this out and it looks like paths that don't match are automatically routed to our HTML index page. This seems wrong--bad `/api` requests should probably get an error response.

Reply:
we could create a catchall `/api` regex and serve 404s or something.

Reply:
Oh also all of these would be better with a $ at the end of the regex.